### PR TITLE
More GC Config Tests

### PR DESCRIPTION
**Pull request by markfields**
Finish what I started in #11226 - MORE TESTS!

### CODE REVIEW

File: packages/runtime/container-runtime/src/test/garbageCollection.spec.ts [hunk 1]
```diff
@@ -97,17 +119,22 @@ describe("Garbage Collection Tests", () => {
             getLastSummaryTimestampMs: () => Date.now(),
         });
     }
+    let gc: GcWithPrivates | undefined;
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/test/garbageCollection.spec.ts:122]: After reading through the PR, I don't understand why you made this change

Comment (markfields) [reply packages/runtime/container-runtime/src/test/garbageCollection.spec.ts:122]: For the `gc?.dispose()` in `afterEach`

File: packages/runtime/container-runtime/src/test/garbageCollection.spec.ts [hunk 2]
```diff
@@ -209,52 +224,221 @@ describe("Garbage Collection Tests", () => {
...
+            });
         });
 
+        describe("Session Behavior (e.g. 'shouldRun' fields)", () => {
+            beforeEach(() => {
+                injectedSettings["Fluid.GarbageCollection.TestOverride.InactiveTimeoutMs"] = 1; // To ensure it's less than sweep timeout
+            });
+
+            describe("shouldRunGC", () => {
+                const testCases: [{ gcEnabled: boolean; disableGC?: boolean; runGC?: boolean; }, boolean /* expectedResult */][] = [
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/test/garbageCollection.spec.ts:359]: I would put this as one object, 
`[{ gcEnabled: boolean; disableGC?: boolean; runGC?: boolean; expectedShouldRunGC: boolean}]`

File: packages/runtime/container-runtime/src/test/garbageCollection.spec.ts [hunk 3]
```diff
@@ -209,52 +224,221 @@ describe("Garbage Collection Tests", () => {
...
+                        injectedSettings[runGCKey] = testCase.runGC;
+                        gc = createGcWithPrivateMembers(undefined /* metadata */, { gcAllowed: testCase.gcEnabled, disableGC: testCase.disableGC });
+                        assert.equal(gc.gcEnabled, testCase.gcEnabled, "PRECONDITION: gcEnabled set incorrectly");
+                        assert.equal(gc.shouldRunGC, expectedResult, "shouldRunGC not set as expected");
+                    });
+                });
+            });
+            describe("shouldRunSweep", () => {
+                const testCases: [{ shouldRunGC: boolean; setSweepTimeout: boolean; sweepEnabled: boolean; runSweep?: boolean; }, boolean /* expectedResult */][] = [
+                    [{ shouldRunGC: false, setSweepTimeout: true, sweepEnabled: true, runSweep: true }, false],
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/test/garbageCollection.spec.ts:378]: Same as above for better readability

File: packages/runtime/container-runtime/src/test/garbageCollection.spec.ts (hunk 2 above)
Comment (markfields) [reply packages/runtime/container-runtime/src/test/garbageCollection.spec.ts:359]: I started that way, but switched to keep the expected value out of the test name (see line 368)

File: packages/runtime/container-runtime/src/test/garbageCollection.spec.ts [hunk 4]
```diff
@@ -209,52 +224,221 @@ describe("Garbage Collection Tests", () => {
...
+                        assert.equal(gc.shouldRunGC, testCase.shouldRunGC, "PRECONDITION: shouldRunGC set incorrectly");
+                        assert.equal(gc.sweepTimeoutMs !== undefined, testCase.setSweepTimeout, "PRECONDITION: sweep timeout set incorrectly");
+                        assert.equal(gc.sweepEnabled, testCase.sweepEnabled, "PRECONDITION: sweepEnabled set incorrectly");
+                        assert.equal(gc.shouldRunSweep, expectedResult, "shouldRunSweep not set as expected");
+                    });
+                });
+            });
+            describe("inactiveTimeoutMs", () => {
+                const testCases: [{ testOverride?: number; option?: number; }, number /* expectedResult */][] = [
+                    [{ testOverride: 123, option: 456 }, 123],
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/test/garbageCollection.spec.ts:400]: Same as above

File: packages/runtime/container-runtime/src/test/garbageCollection.spec.ts [hunk 5]
```diff
@@ -209,52 +224,221 @@ describe("Garbage Collection Tests", () => {
...
+                    injectedSettings["Fluid.GarbageCollection.TestOverride.InactiveTimeoutMs"] = Number.MAX_VALUE;
+                    assert.throws(
+                        () => { gc = createGcWithPrivateMembers(); },
+                        (e) => e.errorType === "usageError",
+                        "inactiveTimeout must not be greater than sweepTimeout");
+                });
+            });
+            describe("testMode", () => {
+                const testCases: [{ setting?: boolean; option?: boolean; }, boolean /* expectedResult */][] = [
+                    [{ setting: true, option: false }, true],
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/test/garbageCollection.spec.ts:422]: same as above

**Review by tyler-cai-microsoft — APPROVED** [review]
Other than some nits, LGTM

**Pull request merged**